### PR TITLE
Clarified `dhcp-identifier` option

### PR DESCRIPTION
**Pull request by lmlsna**
## Description
Added information about the other accepted scalar `duid` which can be specified or is assumed if the option is omitted entirely.

## Checklist

- [x] Runs `make check` successfully.
- [x] Retains 100% code coverage (`make check-coverage`).
- [x] New/changed keys in YAML format are documented.
- [ ] \(Optional\) Adds example YAML for new feature.
- [ ] \(Optional\) Closes an open bug in Launchpad.

### CODE REVIEW

File: doc/netplan.md [hunk 1]
```diff
@@ -285,9 +285,10 @@ Virtual devices
 
 ``dhcp-identifier`` (scalar)
 
-:   When set to 'mac'; pass that setting over to systemd-networkd to use the
-    device's MAC address as a unique identifier rather than a RFC4361-compliant
-    Client ID. This has no effect when NetworkManager is used as a renderer.
+:   (networkd backend only) Sets the source of DHCPv4 client identifier. If "mac"
+    is specified, the MAC address of the link is used. If this option is omitted,
+    or if "duid" is specified, networkd will generate an RFC4361-compliant client
```
Comment (slyon) [inline doc/netplan.md:290]:
```suggestion
:   (networkd backend only) Sets the source of DHCPv4 client identifier. If ``mac``
    is specified, the MAC address of the link is used. If this option is omitted,
    or if ``duid`` is specified, networkd will generate an RFC4361-compliant client
```

**Review by slyon — APPROVED** [review]
Thank you very much for contributing to netplan and clarifying this documentation!

I'd only like to see the possible values emphasized as code-block, as we do at most of the other places in our reference (see inline suggestion). Once this is addressed, I'm +1 on merging!

**Comment by lmlsna** [comment]
Should I squash these commits?

**Comment by slyon** [comment]
Thanks a lot for the update @lmlsna ! No need to squash the commits, as I will squash-merge the full PR anyways.

LGTM. Merging.

**Pull request merged**